src/Core/tests/DeviceTests/Handlers/Entry/EntryHandlerTests.Android.cs

```diff
@@ -15,6 +15,46 @@ namespace Microsoft.Maui.DeviceTests
 {
 	public partial class EntryHandlerTests
 	{
+		[Theory(DisplayName = "Validates Keyboard updates correctly using IsPassword")]
+		[InlineData(nameof(Keyboard.Text), false)]
+		[InlineData(nameof(Keyboard.Numeric), true)]
+		public async Task ValidateKeyboardUpdatesCorrectly(string keyboardName, bool expected)
+		{
+			var layout = new LayoutStub();
+
+			var keyboard = (Keyboard)typeof(Keyboard).GetProperty(keyboardName).GetValue(null);
+
+			var entry = new EntryStub()
+			{
+				IsPassword = false,
+				Keyboard = keyboard,
+				Text = "Password"
+			};
+
+			var button = new ButtonStub
+			{
+				Text = "Change IsPassword"
+			};
+
+			layout.Add(entry);
+			layout.Add(button);
+
+			var clicked = false;
+
+			button.Clicked += delegate
+			{
+				entry.IsPassword = !entry.IsPassword;
+				clicked = true;
+			};
+
+			await PerformClick(button); // IsPassword = true
+			Assert.True(clicked);
+
+			await PerformClick(button); // IsPassword = false
```

Comment:
Wondering if we actually need the button interaction logic - is it enough maybe to just switch IsPassword to true?

Reply:
Yeah, you are right. This is just replicating the scenario from the issue but is not 100% necessary.